ToolBox/IQFeed/IQFeedFileHistoryProvider.cs

```diff
@@ -0,0 +1,207 @@
+            );
+
+            return GetDataFromFile(request, ticker, start, end, isEquity);
+        }
+
+        private IEnumerable<BaseData> GetDataFromFile(HistoryRequest request, string ticker, DateTime startDate, DateTime? endDate, bool isEquity)
+        {
+            try
+            {
+                string filename;
+
+                switch (request.Resolution)
+                {
+                    case Resolution.Tick:
+                        filename = _lookupClient.Historical.File.GetHistoryTickTimeframeAsync(ticker, startDate, endDate, dataDirection: DataDirection.Oldest).SynchronouslyAwaitTaskResult();
+                        return GetDataFromTickMessages(filename, request, isEquity);
+
+                    case Resolution.Daily:
+                        filename = _lookupClient.Historical.File.GetHistoryDailyTimeframeAsync(ticker, startDate, endDate, dataDirection: DataDirection.Oldest).SynchronouslyAwaitTaskResult();
+                        return GetDataFromDailyMessages(filename, request, isEquity);
+
+                    default:
+                        var interval = new Interval(GetPeriodType(request.Resolution), 1);
+                        filename = _lookupClient.Historical.File.GetHistoryIntervalTimeframeAsync(ticker, interval.Seconds, startDate, endDate, dataDirection: DataDirection.Oldest).SynchronouslyAwaitTaskResult();
+                        return GetDataFromIntervalMessages(filename, request, isEquity);
+                }
+            }
+            catch (Exception e)
+            {
+                Log.Error($"IQFeedHistoryProvider.GetDataFromFile(): {e}");
+            }
+
+            return Enumerable.Empty<BaseData>();
+        }
+
+        /// <summary>
+        /// Stream IQFeed TickMessages from disk to Lean Tick
+        /// </summary>
+        /// <param name="filename"></param>
+        /// <param name="request"></param>
+        /// <param name="isEquity"></param>
+        /// <returns>Converted Tick</returns>
+        private IEnumerable<BaseData> GetDataFromTickMessages(string filename, HistoryRequest request, bool isEquity)
+        {
+            // We need to discard ticks which are not impacting the price, i.e those having BasisForLast = O
+            // To get a better understanding how IQFeed is resampling ticks, have a look to this algorithm:
+            // https://github.com/mathpaquette/IQFeed.CSharpApiClient/blob/1b33250e057dfd6cd77e5ee35fa16aebfc8fbe79/src/IQFeed.CSharpApiClient.Extensions/Lookup/Historical/Resample/TickMessageExtensions.cs#L41
+            foreach (var tick in TickMessage.ParseFromFile(filename).Where(t => t.BasisForLast != 'O'))
+            {
+                var time = isEquity ? tick.Timestamp : tick.Timestamp.ConvertTo(TimeZones.NewYork, TimeZones.EasternStandard);
+                yield return new Tick(time, request.Symbol, (decimal)tick.Last, (decimal)tick.Bid, (decimal)tick.Ask)
```

Comment:
Seems the `LookupClient` should be using `decimal` internally instead of double to avoid loss of precision? Not sure if possible

Reply:
Ticks seem to be Quote type but the rest of the data is Trade type, is this right?

Reply:
@Martin-Molinero not really and this is not a problem because prices are like 300000.1234 precision for Berkshire.

Reply:
> Ticks seem to be Quote type but the rest of the data is Trade type, is this right?

Quotes are only available when a trade occur. I did't change anything over there. The results are the same as fair I can remember.  Check OLD vs NEW https://github.com/QuantConnect/Lean/issues/4301

Reply:
Please use decimal for prices, remove commented out code, and remove hardcoded conversion to Eastern standard  ... let's get it right this time.